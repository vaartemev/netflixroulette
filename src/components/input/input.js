import React from 'react';
import './input.scss';

// search__field

export const Input = () => {
  return (
      <div className="wrapinput">
        <input type="text" placeholder="What do you what find?" className="search__field" />
      </div>
  )
}
