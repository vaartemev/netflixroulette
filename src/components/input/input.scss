.search-field{
	&__wrapper {
		&:after {
			content: '';
			display: block;
			height: 2px;
		}

		&:before {
			content: 'â†²';
			z-index: 10;
			font-size: 30px;
			position: absolute;
			right: 20px;
			top: 40%;
			transform: translateY(-50%);
			color: #585858;
		}
		&:focus-within {
			&:after {
				background-color: #f55263;
				animation: border .3s ease-in-out;
			}
		}
	}
}

.search__field {
	display: block;
	width: 100%;
	height: 50px;
	font-size: 20px;
	font-weight: 100;
	font-family: 'Roboto', sans-serif;
	border: none;
	background-color: #333333;
	color: #ffffff;
	text-indent: 10px;
	margin-top: 10px;
	transition: .4s;
	position: relative;
}

@keyframes border {
	from {
		background-color: #fff;
		width: 0%;
	}

	to {
		background-color: #f55263;
		width: 100%;
	}
}