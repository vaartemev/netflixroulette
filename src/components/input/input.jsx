import React from 'react';
import './input.scss';

export const Input = ({ searchValue, setValue, handleOnKeyPress }) => {
  return (
    <label className="search__field--wrapper">
      <input
        type="text"
        placeholder="What do you what find?"
        className="search__field"
        onChange={e => setValue((searchValue = e.target.value))}
        onKeyPress={e =>
          e.key === 'Enter' ? handleOnKeyPress(searchValue) : false
        }
      />
    </label>
  );
};
