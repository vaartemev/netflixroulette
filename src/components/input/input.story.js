import React from 'react';
import { storiesOf } from '@storybook/react';
import { Input } from './input';

storiesOf('Input', module).add('Input', () => <Input />);
