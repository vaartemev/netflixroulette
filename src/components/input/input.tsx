import React from 'react';
import './input.scss';

interface Props {
  searchValue: string | string[];
  placeholderText: string;
  handleOnEnterPress: (searchValue: string | string[], key: string) => void;
  handleOnInput: (value: string) => void;
}

export const Input = ({
  searchValue,
  placeholderText,
  handleOnEnterPress,
  handleOnInput,
}: Props) => {
  return (
    <label className="search-field__wrapper" htmlFor="input">
      <input
        type="text"
        placeholder={placeholderText}
        className="search__field"
        onInput={({ currentTarget: { value } }) => handleOnInput(value)}
        onKeyDown={({ key }) => handleOnEnterPress(searchValue, key)}
      />
    </label>
  );
};
