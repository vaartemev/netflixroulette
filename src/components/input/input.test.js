import { shallow } from 'enzyme';
import { Input } from './input';

describe('<Input />', () => {
  const props = {
    placeholderText: 'some text'
  };

  const output = shallow(<Input placeholderText={props.placeholderText} />);

  it('Should be rendering with placeholder "some text"', () => {
    const input = output.find('input');
    expect(input.prop('placeholder')).toEqual(props.placeholderText);
  });

  it('Render property', () => {
    expect(output).toMatchSnapshot();
  });
});
