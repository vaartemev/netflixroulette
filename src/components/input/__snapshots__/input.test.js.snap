// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> Render property 1`] = `ShallowWrapper {}`;
